docs/source/x_ext.rst

```diff
@@ -531,13 +503,83 @@ The |processor| shall cause an illegal instruction fault when attempting to exec
+  | ``register``              | x_register_t    | output          | Register packet.                                                                                                             |
+  +---------------------------+-----------------+-----------------+------------------------------------------------------------------------------------------------------------------------------+
+
+:numref:`Register type` describes the ``x_register_t`` type.
+
+.. table:: Register type
+  :name: Register type
+
+  +------------------------+--------------------------+-----------------------------------------------------------------------------------------------------------------+
+  | **Signal**             | **Type**                 | **Description**                                                                                                 |
+  +------------------------+--------------------------+-----------------------------------------------------------------------------------------------------------------+
+  | ``id``                 | :ref:`id_t <id>`         | Identification of the offloaded instruction.                                                                    |
+  +------------------------+--------------------------+-----------------------------------------------------------------------------------------------------------------+
+  | ``rs[X_NUM_RS-1:0]``   | logic [X_RFR_WIDTH-1:0]  | Register file source operands for the offloaded instruction.                                                    |
+  +------------------------+--------------------------+-----------------------------------------------------------------------------------------------------------------+
+  | ``rs_valid``           | :ref:`registerflags_t    | Validity of the register file source operand(s). If register pairs are supported, the validity is signaled for  |
+  |                        | <registerflags>`         | each register within the pair individually.                                                                     |
+  +------------------------+--------------------------+-----------------------------------------------------------------------------------------------------------------+
+  | ``ecs``                | logic [5:0]              | Extension Context Status ({``mstatus.xs``,``mstatus.fs``,``mstatus.vs``}).                                      |
+  +------------------------+--------------------------+-----------------------------------------------------------------------------------------------------------------+
+  | ``ecs_valid``          | logic                    | Validity of the Extension Context Status.                                                                       |
+  +------------------------+--------------------------+-----------------------------------------------------------------------------------------------------------------+
+
+There are two main scenarios, in how the register interface will be used:
```

Comment:
This will seemingly lead to two incompatible coprocessor types: One that is allowed to put back pressure on the register interface and one that is not allowed to do that? Can we avoid somehow that we end up with incompatible coprocessors? If in total we get 2 flavors of coprocessors (low-end, high-end), then this is maybe acceptable; if we are going to add more flavors then we are causing fragmentation issues.

Reply:
it is a topic we need to address. its on the top of my list.

Reply:
I suggest we will address this in a separte PR